result_mask_test.go

```diff
@@ -1,17 +1,26 @@
 package floc
 
-import "testing"
+import (
+	"testing"
+)
 
 func TestEmptyResultMask(t *testing.T) {
 	set := EmptyResultMask()
-
-	for _, r := range []Result{None, Completed, Canceled, Failed} {
+	results := []Result{None, Completed, Canceled, Failed}
+	length := len(results)
+	maskeds := make(chan bool, length)
+	for _, r := range results {
 		go func(r Result) {
-			if set.IsMasked(r) {
-				t.Fatalf("%s expects %s to be not masked", t.Name(), r.String())
-			}
+			maskeds <- set.IsMasked(r)
 		}(r)
 	}
+
+	for i := 0; i < length; i++ {
+		flag := <-maskeds
+		if flag {
+			t.Fatalf("%s expects %s to be not masked", t.Name(), results[i].String())
```

Comment:
This may work improperly because the order goroutines ran earlier is not predictable. For example, `Failed` may be tested and written to the channel earlier than `Canceled`.

I suggest instead of using `chan bool` use `chan int` where we write `-1` if set is not masked or the index of the Result variant which test failed.

```golang
for i, r := range results {
	go func(n int, r Result) {
		if set.IsMasked(r) {
			maskeds <- n
		} else {
			maskeds <- -1
		}
	}(i, r)
}
```

and then

```golang
for i := 0; i < length; i++ {
	index := <-maskeds
	if index >= 0 {
		t.Fatalf("%s expects %s to be not masked", t.Name(), results[index].String())
	}
}
```

Reply:
Ya, correct. Sorry for missing out this, already amended.

Reply:
Yeah I have Noone that would help and everyone who said they were were only trying to do the wrong stuff only to laugh at me